react/__mocks__/SubsMockComponent.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { FunctionComponent } from 'react'
+import { InjectedIntlProps, injectIntl } from 'react-intl'
+import { getSubscriptionInfo } from '../utils'
+
+const SubsMockComponent: FunctionComponent<
+  { attachmentItem: Attachment } & InjectedIntlProps
+> = ({ attachmentItem, intl }) => {
+  const {
+    isSubscription,
+    subsFrequency,
+    subsPurchaseDay,
+    subsValidityBegin,
+    subsValidityEnd,
+  } = getSubscriptionInfo(attachmentItem, intl)
+
+  return (
+    <ul>
```

Comment:
You can change it a render props component


```js

return children({
    isSubscription,
    subsFrequency,
    subsPurchaseDay,
    subsValidityBegin,
    subsValidityEnd,
  })
```

Then, at the tests you may pass a mock function and check the params that were called to it:


```js
const fn = jest.mock()

render(
<SubscriptionMock
  attachmentItem={attachmentItem}
  children={fn} />
)

const params = fn.mock.calls[0][0]

expected(params.isSubscription).toBe(true)
expected(params.subsFrequency).toBe('foo')
// etc
```

Reply:
Yep! Much cleaner hahahah
Just one thing, I could not return `children({...})` directly, so I called the mock function from inside the mock component and after that I return `null`, jest wouldn't let me return "nothing" in my component, as it assumed `fn` was.